API.md

```diff
@@ -1,168 +1,345 @@
+                    "enabled": {
+                        "defaultValue": "false",
+                        "description": "Record event",
+                        "name": "Enabled"
+                    },
+                    "stackTrace": {
+                        "defaultValue": "false",
+                        "description": "Record stack traces",
+                        "name": "Stack Trace"
+                    },
+                    "threshold": {
+                        "defaultValue": "0ns[ns]",
+                        "description": "Record event with duration above or equal to threshold",
+                        "name": "Threshold"
+                    }
+                },
+                "typeId": "jdk.SocketWrite"
+            },
+            {
+                "category": [
+                    "Java Application"
+                ],
+                "description": "Reading data from a socket",
+                "name": "Socket Read",
+                "options": {
+                    "enabled": {
+                        "defaultValue": "false",
+                        "description": "Record event",
+                        "name": "Enabled"
+                    },
+                    "stackTrace": {
+                        "defaultValue": "false",
+                        "description": "Record stack traces",
+                        "name": "Stack Trace"
+                    },
+                    "threshold": {
+                        "defaultValue": "0ns[ns]",
+                        "description": "Record event with duration above or equal to threshold",
+                        "name": "Threshold"
+                    }
+                },
+                "typeId": "jdk.SocketRead"
+            }
+        ],
+        "port": 9093,
         "target": {
             "apiVersion": "v1",
             "kind": "Service",
-            "name": "containerjfr",
+            "name": "jmx-listener",
             "namespace": "default",
-            "resourceVersion": "298899",
-            "uid": "f895e203-37e9-11ea-8866-52fdfc072182"
+            "resourceVersion": "392780",
+            "uid": "4f5b0695-6fac-11ea-ae0c-52fdfc072182"
         }
     }
 }
 ```
+(Event listing abbreviated for readability)
 
 ## Creating a new Flight Recording
 
-To start a new recording, you will need to add a new recording request to the `FlightRecorder` in the `spec.recordingRequests` array. You can do this nicely on the command line with `oc edit`. The recording request must include the following:
+To start a new recording, you will need to create a new `Recording` custom resource. The `Recording` must include the following:
 
-1. `name`: a string uniquely identifying the recording within that service
+1. `name`: a string uniquely identifying the recording within that service.
 2. `eventOptions`: an array of string options passed to Container JFR. The `"ALL"` special string can be used to enable all available events.
```

Comment:
I think this needs to be updated - since ContainerJFR gained support for remote target event templates, the event specifier string syntax for that is "template=Foo". The `ALL` special string was made into a pseudo-template, so it should be selected with `template=ALL`.

Reply:
Understood, I've changed this now. I'll have to add some support for templates to the API in the future as well.